### PR TITLE
Fix renaming a dir with sub-directories

### PR DESCRIPTION
**Pull request by nono**
I think it should fix issue #141

### CODE REVIEW

File: memmap_test.go [hunk 1]
```diff
@@ -449,3 +449,47 @@ func TestMemFsUnexpectedEOF(t *testing.T) {
...
+	const dstPath = "/dst"
+	const subDir = "dir"
+
+	fs := NewMemMapFs()
+
+	err := fs.MkdirAll(srcPath+FilePathSeparator+subDir, 0777)
+	if err != nil {
+		t.Errorf("MkDirAll failed: %s", err)
+		return
+	}
```
Comment (kargakis) [inline memmap_test.go:464]: Can you also add a file in the subdirectory and not have only directories in the test?

File: memmap_test.go [hunk 2]
```diff
@@ -449,3 +449,47 @@ func TestMemFsUnexpectedEOF(t *testing.T) {
...
+
```
Comment (kargakis) [inline memmap_test.go:465]: Also please make sure to stat the file here before renaming

File: memmap.go [hunk 3]
```diff
@@ -270,7 +270,7 @@ func (m *MemMapFs) RemoveAll(path string) error {
 	defer m.mu.RUnlock()
 
 	for p := range m.getData() {
-		if strings.HasPrefix(p, path) {
+		if strings.HasPrefix(p, path+FilePathSeparator) {
```
Comment (kargakis) [inline memmap.go:273]: Why is this change needed?

**Comment by nono** [comment]
@bep hi, this is an old pull request but I think it would still be a good fix for afero. I have rebased it and listened to the comments made by @0xmichalis. If you have time to look at it, it would be appreciated.

**Comment by nono** [comment]
Thanks :heart:

**Comment by hanagantig** [comment]
It doesn't fix the issue https://github.com/spf13/afero/issues/141
I have described the problem here https://github.com/spf13/afero/issues/141#issuecomment-1530047212

You can check my PR which fix it https://github.com/spf13/afero/pull/364
